modules/database/export.bal

```diff
@@ -316,88 +308,94 @@ public isolated transactional function getExperimentDBExport(int experimentId) r
+        var path = check file:joinPath(storageLocation, experimentId.toString(), experimentData.location);
+        var abspath = check file:getAbsolutePath(path);
+        log:printDebug("Add file " + abspath + "..."); // TODO: move to debug
+        dataFileLocations.push(abspath);
     }
 
-    // string tmpDir = check file:createTempDir();
-    if !(check file:test("tmp", file:EXISTS)) {
-        check file:createDir("tmp", file:RECURSIVE);
-    }
+    var tmpDirBase = getTmpDir(os);
+    var tmpDir = check file:joinPath(tmpDirBase, "export-" + exportId.toString());
+    tmpDir = check ensureDirExists(tmpDir);
     json experimentCompleteJson = experimentComplete;
-    string jsonFile = "tmp/experiment.json";
+    string jsonFile = check file:joinPath(tmpDir, "experiment.json");
     var jsonPath = check file:getAbsolutePath(jsonFile);
     if check file:test(jsonPath, file:EXISTS) {
         check file:remove(jsonPath);
     }
-    log:printInfo("Write " + jsonPath + " ...");
+    log:printInfo("Write " + jsonPath + " ..."); // TODO: change to debug
     check io:fileWriteJson(jsonPath, experimentCompleteJson);
 
     // create zip-  add all files (experiment file(s) + data files) to ZIP
-    string zipFileName = regex:replaceAll(experimentComplete.experiment.name, "\\s+", "-") + ".zip";
-    var zipPath = check file:getAbsolutePath("tmp/" + zipFileName);
-    log:printInfo("Create zip " + zipPath + " ...");
+    string zipFileName = regex:replaceAll(experimentComplete.experiment.name, "[\\s+\\\\/:<>\\|\\?\\*]", "-") + ".zip";
+    var zipPath = check file:joinPath(tmpDir, zipFileName);
+    var zipPathAbs = check file:getAbsolutePath(zipPath);
+    log:printInfo("Create zip " + zipPathAbs + " ..."); // TODO: change to debug
+
+    check zipExperiment(zipPathAbs, jsonPath, dataFileLocations, os);
+
+    ExperimentExportZip exportResult = {name: zipFileName, location: zipPathAbs};
+    return exportResult;
+}
 
+# Zip all created export files into one zip file
+#
+# + zipPath - path of zip
+# + jsonPath - path of experiment json
+# + dataFileLocations - paths for data files 
+# + os - os type to determine appropriate exec command
+# + return - error
+public isolated transactional function zipExperiment(string zipPath, string jsonPath, string[] dataFileLocations, string os) returns error? {
     // add experiment.json
     os:Process result;
-    if os == "linux" {
-        result = check os:exec({value: "zip", arguments: ["-j", zipPath, jsonPath]});
-    } else if os == "windows" {
+    if os.toLowerAscii().includes("windows") {
         result = check os:exec({value: "powershell", arguments: ["Compress-Archive", "-Update", jsonPath, zipPath]});
     } else {
-        return error("Unsupported operating system! At the moment, we support 'linux' and 'windows' for importing/exporting experiments. Please make sure to properly specify the os env var or config entry.");
+        if !os.toLowerAscii().includes("linux") {
```

Comment:
see comment above, also check if else needs `{}` if it is meant as else-if.

Reply:
in case it's not linux but linux-like it may be worth printing an error and still trying to run the exec. If it fails an exception should be raised. Therefore the if within the else.